### PR TITLE
Docs: Organize meta and describe visitor in Working with Rules

### PR DESCRIPTION
**Pull request by pedrottimark**
* bulleted list of rule files so it would be easy to add rule doc
* bulleted lists of meta properties (see #5417) and illustrated them in the first source file example
* use terms visit (verb) and visitor (noun) discussed in #5843 @jfmengels Can you please critique?
* use terms event and handler for code path analysis @mysticatea Can you please critique?
* illustrate visitors and handlers from `array-callback-return` in the second source file example

### CODE REVIEW

File: docs/developer-guide/working-with-rules.md [hunk 1]
```diff
@@ -1,69 +1,70 @@
...
-
+    // declare the state of the rule
     return {
-
-        "Identifier:exit": function(node) {
-            // do something with node
+        ReturnStatement: function(node) {
+            // at a ReturnStatement node while going down
+        },
+        "FunctionExpression:exit":
```
Comment (platinumazure) [inline docs/developer-guide/working-with-rules.md]: Am I missing something? This property needs a value (there is no fallthrough of object key/value pairs).

Comment (pedrottimark) [inline docs/developer-guide/working-with-rules.md]: @platinumazure You will smile. I woke up this morning thinking how I left out the `case` keywords at the fallthrough for function expressions, and then, oh noooo!

Comment (platinumazure) [inline docs/developer-guide/working-with-rules.md]: It's okay, we all have days like that. Good work on these docs!

**Comment by jfmengels** [comment]
LGTM :)

**Comment by nzakas** [comment]
Waiting on feedback from @mysticatea

**Comment by pedrottimark** [comment]
Motivated by discussion in chat, added under Applying Fixes an **Important** note about `meta.fixable` property with a link to Rule Basics.

**Comment by nzakas** [comment]
Lgtm. Would rather merge in now and update later if we find problems.

**Pull request merged**